pkg/wallet/api.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package wallet
+
+// Wallet interface
+type Wallet interface {
+	Crypto
+	Pack
+}
+
+// Crypto interface
+type Crypto interface {
+
+	// CreateSigningKey create a new public/private signing keypair.
+	//
+	// Returns:
+	//
+	// KeyInfo: A `KeyInfo` representing the keypair
+	//
+	// error: error
+	CreateSigningKey(metadata map[string]string) (KeyInfo, error)
+
+	// GetSigningKey Fetch info for a signing keypair.
+	//
+	// Args:
+	//
+	// verkey: The verification key of the keypair
+	//
+	// Returns:
+	//
+	// KeyInfo: A `KeyInfo` representing the keypair
+	//
+	// error: error
+	GetSigningKey(verKey string) (KeyInfo, error)
+
+	// SignMessage sign a message using the private key associated with a given verification key.
+	//
+	// Args:
+	//
+	// message: The message to sign
+	//
+	// fromVerKey: Sign using the private key related to this verification key
+	//
+	// Returns:
+	//
+	// []byte: The signature
+	//
+	// error: error
+	SignMessage(message []byte, fromVerKey string) ([]byte, error)
+
+	// DecryptMessage decrypt message
+	//
+	// Args:
+	//
+	// encMessage: The encrypted message content
+	//
+	// toVerKey:The verification key of the recipient.
+	//
+	// []byte: Decrypted message content
+	//
+	// string: The sender verification key
+	//
+	// error: error
+	DecryptMessage(encMessage []byte, toVerKey string) ([]byte, string, error)
+}
+
+// Pack provide methods to pack and unpack msg
+type Pack interface {
```

Comment:
I think slim interfaces naming should be action oriented (e.g., with an er suffix).